src/components/dashboard/DashboardChat.tsx

```diff
@@ -1,268 +1,29 @@
-      );
-
-      eventSource.onmessage = (event) => {
-        try {
-          console.log("Raw SSE data:", event.data);
-          if (!event.data || event.data.trim() === "") {
-            console.log("Received empty SSE event, skipping");
-            return;
-          }
-
-          const data = JSON.parse(event.data);
-          if (!data || typeof data !== "object") {
-            console.warn("Parsed data is not an object:", data);
-            return;
-          }
-
-          switch (data.type) {
-            case "step":
-            case "task":
-            case "result":
-              setStreamingMessages((prev) => [
-                ...prev,
-                { type: data.type, content: data.content },
-              ]);
-              break;
-            default:
-              console.warn("Unknown message type:", data.type);
-          }
-
-          // Close the connection quietly if a "result" message is received
-          if (data.type === "result") {
-            eventSource.close();
-            setIsLoading(false);
-          }
-        } catch (error) {
-          console.error("Error parsing SSE data:", error);
-          console.error("Raw data that failed to parse:", event.data);
-        }
-      };
-
-      // Handle errors but only show if it's an unexpected failure
-      eventSource.onerror = (error) => {
-        if (isLoading) {
-          console.warn("SSE connection closed after completion");
-          setIsLoading(false);
-        } else {
-          console.error("EventSource failed:", error);
-          toast({
-            title: "Connection Failed",
-            description:
-              "There was a problem with the connection. Please try again.",
-            variant: "destructive",
-          });
-        }
-      };
-
-      eventSource.addEventListener("finish", () => {
-        eventSource.close();
-        setIsLoading(false);
-
-        const assistantMessage: Message = {
-          role: "assistant",
-          content: "Task completed",
-          timestamp: new Date(),
-          streamMessages: streamingMessages,
-        };
-        setMessages((prev) => [...prev, assistantMessage]);
-        setStreamingMessages([]);
-      });
-    } catch (error) {
-      console.error("Error sending message:", error);
-      toast({
-        title: "Error",
-        description:
-          error instanceof Error ? error.message : "Failed to send message",
-        variant: "destructive",
-      });
-      setIsLoading(false);
-    }
-  };
+export default function DashboardChat() {
+  const {
+    messages,
+    input,
+    setInput,
+    isLoading,
+    handleSubmit,
+    handleResetHistory,
+    messagesEndRef,
+  } = useChat();
 
   return (
     <Card className="w-full">
       <CardContent className="space-y-4 p-4">
-        <div className="h-[300px] overflow-y-auto space-y-4">
+        <div className="h-[550px] overflow-y-auto space-y-4">
```

Comment:
We should experiment with 90% or 100% here for height since the layout is all chat now.